Improve Datadog integration docs (#5261)
solo-io/gloo: docs/content/guides/integrations/datadog/_index.md

### Install Kubernetes and Envoy Integrations

> From the Datadog portal, select [Integrations](https://app.datadoghq.com/account/settings#integrations) and install the Envoy and Kubernetes integrations.

Reviewer: ```suggestion
From the Datadog portal, select [**Integrations**](https://app.datadoghq.com/account/settings#integrations). Then, install the **Envoy** and **Kubernetes** integrations.
```

### Retrieve the API key

> From the Datadog portal, select Integrations and the Agents subcategory. Then select the [Kubernetes agent](https://app.datadoghq.com/account/settings#agent/kubernetes). Under the directions, you can find the Helm command to install the Datadog agent on your cluster. Your API key will be included in the sample command. It will look like this:

Reviewer: ```suggestion
From the Datadog portal, select **Integrations > Agents**. Then, select the [**Kubernetes** agent](https://app.datadoghq.com/account/settings#agent/kubernetes). In the directions, find the Helm command to install the Datadog agent on your cluster. Your API key is included in the sample command, similar to the following example.
```

These annotations can also be added declaratively via helm, for example if using Gloo Edge Enterprise, these annotations can be added as a value for `gloo.gatewayProxies.gatewayProxy.podTemplate.extraAnnotations`.

{{< notice note >}}
If you upgrade the cluster using Helm version 3, these annotations should stay in place. Helm 3 uses a three-way merge when performing an update. Helm version 2 will also attempt a merge, but may have issues with changes made using kubectl edit. You should update the values used by Helm to include these annotations. Note that Helm 2 is not supported in Gloo Edge v1.8.0 and later.
{{< /notice >}}

> More information on Helm three-way merges is available [here](https://helm.sh/docs/faq/changes_since_helm2/#improved-upgrade-strategy-3-way-strategic-merge-patches).

Reviewer: ```suggestion
For more information about merging strategies, see the [Helm documentation](https://helm.sh/docs/faq/changes_since_helm2/#improved-upgrade-strategy-3-way-strategic-merge-patches).
```
